src/bin/hal/cmd/psbt.rs

```diff
@@ -496,29 +483,14 @@ fn exec_rawsign<'a>(matches: &clap::ArgMatches<'a>) {
 	if i >= psbt.inputs.len() {
 		panic!("PSBT input index out of range")
 	}
-	let redeem_script = psbt.inputs[i].redeem_script.as_ref().map(|x|
-		bitcoin::blockdata::script::Builder::new()
-		.push_slice(x.as_bytes())
-		.into_script());
-	let witness_script = psbt.inputs[i].witness_script.as_ref()
-		.map(|x| vec![x.clone().into_bytes()]);
-	let witness = witness_script.unwrap_or(Vec::new());
-	let script_sig = redeem_script.unwrap_or(bitcoin::Script::new());
-
-	let (msg, sighashtype) = {
-		let (spk, amt) = get_spk_amt(&psbt, i);
-		// Call with age and height 0.
-		// TODO: Create a method to rust-bitcoin psbt that outputs sighash
-		// Workaround using miniscript interpreter
-		let interp = miniscript::Interpreter::from_txdata(spk, &script_sig, &witness, 0, 0)
-			.expect("Witness/Redeem Script is not a Miniscript");
-		let sht = psbt.inputs[i].sighash_type.unwrap_or(bitcoin::SigHashType::All);
-		let msg = interp.sighash_message(&psbt.global.unsigned_tx, i, amt, sht);
-		(msg, sht)
-	};
+
+	let tx =  psbt.clone().extract_tx();
+	let mut cache = bitcoin::util::sighash::SighashCache::new(&tx);
+	let msg = psbt.sighash_msg(i, &mut cache, None)
+		.expect("error computing sighash message on psbt");
```

Comment:
nit: I would move this section (because it has no user-based error paths) closer to the `match msg` below until after handling the key to sign.

Reply:
Done.